backends/events/sdl/sdl-events.cpp

```diff
@@ -205,40 +205,46 @@ bool SdlEventSource::handleKbdMouse(Common::Event &event) {
 
 		if (_km.x_vel || _km.y_vel) {
 			if (_km.x_down_count) {
-				if (curTime > _km.x_down_time + _km.delay_time * 12) {
+				if (curTime > _km.x_down_time + 300) {
 					if (_km.x_vel > 0)
-						_km.x_vel+=_km.multiplier;
+						_km.x_vel += _km.multiplier;
 					else
-						_km.x_vel-=_km.multiplier;
-				} else if (curTime > _km.x_down_time + _km.delay_time * 8) {
+						_km.x_vel -= _km.multiplier;
+				} else if (curTime > _km.x_down_time + 200) {
 					if (_km.x_vel > 0)
 						_km.x_vel = 5 * _km.multiplier;
 					else
 						_km.x_vel = -5 * _km.multiplier;
 				}
 			}
 			if (_km.y_down_count) {
-				if (curTime > _km.y_down_time + _km.delay_time * 12) {
+				if (curTime > _km.y_down_time + 300) {
 					if (_km.y_vel > 0)
 						_km.y_vel += _km.multiplier;
 					else
 						_km.y_vel -= _km.multiplier;
-				} else if (curTime > _km.y_down_time + _km.delay_time * 8) {
+				} else if (curTime > _km.y_down_time + 200) {
 					if (_km.y_vel > 0)
 						_km.y_vel = 5 * _km.multiplier;
 					else
 						_km.y_vel = -5 * _km.multiplier;
 				}
 			}
+
 			// The modifier key makes the mouse movement slower
+			// The extra factor of delay/25 makes velocities 
+			// independent of kbdMouse update rate
+			// all velovities were originally chosen
+			// at a delay of 25, so that is the reference
+			// operator order is important to avoid overflow
 			if (_km.modifier) {
-				_km.x += _km.x_vel / 10;
-				_km.y += _km.y_vel / 10;
+				_km.x += ((_km.x_vel / 10) * ((int16)_km.delay_time)) / 25;
+				_km.y += ((_km.y_vel / 10) * ((int16)_km.delay_time)) / 25;
 			} else {
-				_km.x += _km.x_vel;
-				_km.y += _km.y_vel;
+				_km.x += (_km.x_vel * ((int16)_km.delay_time)) / 25;
```

Comment:
This ensures velocities are not twice as fast although delay is now 12 instead of 25